components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -4300,6 +4300,14 @@ public Response createAPIRevision(String apiId, APIRevisionDTO apIRevisionDTO, M
 
             //validate if api exists
             APIInfo apiInfo = validateAPIExistence(apiId);
```

Comment:
can't we include advertise only true into APIInfo

Reply:
Cannot add to this object since this is the object retrieved from the DB